SharedContent/cs-winui-template/WinUI3TemplateCs/MainPage.xaml.cs

```diff
@@ -0,0 +1,156 @@
+﻿using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Navigation;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WinUI3TemplateCs
+{
+
+    public sealed partial class MainPage : Page
+    {
+        public static MainPage Current;
+
+        public List<Scenario> Scenarios
+        {
+            get { return this.scenarios; }
+        }
+
+        public MainPage()
+        {
+            InitializeComponent();
+
+            // This is a static public property that allows downstream pages to get a handle to the MainPage instance
+            // in order to call methods that are in this class.
+            Current = this;
+        }
+
+        public void NotifyUser(string strMessage, InfoBarSeverity severity, bool isOpen = true)
+        {
+            // If called from the UI thread, then update immediately.
+            // Otherwise, schedule a task on the UI thread to perform the update.
+            if (DispatcherQueue.HasThreadAccess)
+            {
+                UpdateStatus(strMessage, severity, isOpen);
+            }
+            else
+            {
+                DispatcherQueue.TryEnqueue(() =>
```

Comment:
Your pattern here requires repeating the same code in two blocks. While it's only one line (for now), it's an unfortunate pattern that may be copied where it's even worse.

I wonder how folks would feel about using `JoinableTaskFactory` in this template. If you did, this entire method body could be replaced with just this:

```cs
await App.JoinableTaskFactory.SwitchToMainThreadAsync();
UpdateStatus(strMessage, severity, isOpen);
```

Reply:
@marb2000 do you have thoughts on switching from `DispatcherQueue` to `JoinableTaskFactory`?

Reply:
From offline thread we will continue using DQ